src/modules/paymentProcessor/PP_Queue_v1.sol

```diff
@@ -0,0 +1,740 @@
+        _queue[client_].removeId(prevId, orderId_);
+    }
+
+    /// @dev    Validates token balance and allowance for a payment.
+    /// @param  token_ Token to check.
+    /// @param  client_ Client address.
+    /// @param  amount_ Amount to check.
+    /// @return valid_ True if balance and allowance are sufficient.
+    function _validTokenBalance(address token_, address client_, uint amount_)
+        internal
+        view
+        returns (bool valid_)
+    {
+        IERC20 token = IERC20(token_);
+        return token.balanceOf(client_) >= amount_
+            && token.allowance(client_, address(this)) >= amount_;
+    }
+
+    /// @notice	Used to claim the unclaimable amount of a particular
+    /// paymentReceiver for a given payment client.
+    /// @param	client Address of the payment client.
+    /// @param	token Address of the payment token.
+    /// @param	paymentReceiver Address of the paymentReceiver for which
+    /// the unclaimable amount will be claimed.
+    function _claimPreviouslyUnclaimable(
+        address client,
+        address token,
+        address paymentReceiver
+    ) internal {
+        // Copy value over.
+        uint amount =
+            _unclaimableAmountsForRecipient[client][token][_msgSender()];
+        // Delete the field.
+        delete _unclaimableAmountsForRecipient[client][token][_msgSender()];
+
+        // Make sure to let paymentClient know that amount doesnt have
+        // to be stored anymore.
+        IERC20PaymentClientBase_v1(client).amountPaid(token, amount);
+
+        // Call has to succeed otherwise no state change.
+        IERC20(token).safeTransferFrom(client, paymentReceiver, amount);
+
+        emit TokensReleased(paymentReceiver, address(token), amount);
+    }
+
+    /// @dev    Validates if a queue ID is valid.
+    /// @param  queueId_ The queue ID to validate.
+    /// @return isValid_ True if the queue ID is valid.
+    function _validQueueId(uint queueId_)
+        internal
+        view
+        returns (bool isValid_)
+    {
+        // Queue ID must be less than or equal to total orders
+        // and greater than 0 (we start from 1).
+        return queueId_ > 0 && queueId_ <= _nextOrderId;
+    }
+
+    /// @dev    Gets payment queue ID from flags and data.
+    /// @param  flags_ The payment order flags.
+    /// @param  data_ Additional payment order data.
+    /// @return queueId_ The queue ID from the data.
+    function _getPaymentQueueId(bytes32 flags_, bytes32[] memory data_)
+        internal
+        pure
+        returns (uint queueId_)
+    {
+        // Check if orderID flag is set (bit 0)
+        bool hasOrderId = uint(flags_) & (1 << FLAG_ORDER_ID) != 0;
+        if (hasOrderId) {
+            if (data_.length > FLAG_ORDER_ID) {
+                // bounds check
+                queueId_ = uint(data_[FLAG_ORDER_ID]);
+            }
+        }
+    }
+
+    /// @dev    Validate uint total amount input.
```

Comment:
Should be a `notice` tag. Also "uint total amount" sounds a bit weird.

Reply:
changed